src/DataTable/DataViewToggle.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { useContext, useState } from 'react';
+import {
+  DataTableContext, Icon, IconButtonToggle, IconButtonWithTooltip,
+} from '..';
+import { GridView, ListView } from '../../icons';
+
+const DataViewToggle = () => {
+  const {
+    dataViewToggleOptions: {
+      isDataViewToggleEnabled,
+      onDataViewToggle,
+      defaultActiveStateValue,
+    },
+  } = useContext(DataTableContext);
+
+  if (!isDataViewToggleEnabled) { return null; }
+
+  const [activeValue, setActiveValue] = useState(defaultActiveStateValue || 'card');
+  const handleOnChange = value => {
+    setActiveValue(value);
+    onDataViewToggle(value);
```

Comment:
nit: Since this is coming in via an object now, should we be defensive around when `onDataViewToggle` is not provided, e.g.:

```jsx
if (onDataViewToggle) {
  onDataViewToggle(value);
}
```

Reply:
hmm indeed, great catch